temporary quarantine directory is not created by plugin queue/quarantine

### system info
Haraka | Haraka.js — Version: 2.8.16
 --- | :--- 
Node | v6.11.3
OS | Linux localhost 3.10.0-693.el7.x86_64 #1 SMP Tue Aug 22 21:09:42 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 1.0.2k-fips  26 Jan 2017

### Expected behavior
If queue/quarantine is loaded it expected to create temporary dir it uses. There is a function init_quarantine_dir in plugins/queue/quarantine.js

### Observed behavior
This directory is not created and I see errors like this in logs:
`haraka[11169]: [ERROR] [1F77F2B6-9EF0-4B84-94D9-D0AAB71688B1.1] [queue/quarantine] Error writing quarantine file: ENOENT: no such file or directory, open '/var/spool/haraka/quarantine/tmp/1F77F2B6-9EF0-4B84-94D9-D0AAB71688B1.1'`

### Steps to reproduce
* add queue/quarantine to config/plugins
* use some other plugin which tries to quarantine messages

Code in the plugins/queue/quarantine.js look suspicious: hook_init_master first defined as
```javascript
exports.hook_init_master = function (next, server) {
    this.init_quarantine_dir(next);
}
```
and then as:
```javascript
exports.hook_init_master = function (next) {
    // At start-up; delete any files in the temporary directory
    .....skipped....
    return next();
};
```
I think exports.hook_init_master should be assigned only once.

--- a/Changes.md
+++ b/Changes.md
@@ -5,6 +5,7 @@
 * Fixes
     * Fix ability to set log level to emerg #2128
     * outbound/hmail: use Buffer to correctly read binary file data + tests #2231
+    * quarantine: consolidate 2x hook_init_master functions
 * Changes
     * outbound/hmail: add JSON sanity test before JSON.parse #2231
     * outbound/index: use newer Buffer.from syntax #2231

--- a/plugins/queue/quarantine.js
+++ b/plugins/queue/quarantine.js
@@ -14,7 +14,9 @@ exports.register = function () {
 };
 
 exports.hook_init_master = function (next, server) {
-    this.init_quarantine_dir(next);
+    this.init_quarantine_dir(() => {
+        this.clean_tmp_directory(next);
+    });
 }
 
 exports.load_quarantine_ini = function () {
@@ -32,7 +34,7 @@ const zeroPad = exports.zeroPad = function (n, digits) {
     return n;
 };
 
-exports.hook_init_master = function (next) {
+exports.clean_tmp_directory = function (next) {
     // At start-up; delete any files in the temporary directory
     // NOTE: This is deliberately syncronous to ensure that this
     //       is completed prior to any messages being received.
@@ -46,7 +48,7 @@ exports.hook_init_master = function (next) {
             fs.unlinkSync(path.join(tmp_dir, dirent[i]));
         }
     }
-    return next();
+    next();
 };
 
 function wants_quarantine (connection) {

